DNN Platform/Website/Default.aspx.cs

```diff
@@ -209,6 +209,19 @@ protected override void OnInit(EventArgs e)
         {
             base.OnInit(e);
 
+            // Set Persian culture to support localization when culture is fa-IR (e.g. for 404 pages)
+            if (this.PortalSettings.CultureCode == "fa-IR")
+            {
+                var newCulture = Services.Localization.Persian.PersianController.GetPersianCultureInfo();
+                System.Threading.Thread.CurrentThread.CurrentUICulture = newCulture;
+            }
+
+            // Add 'rtl' class to body for right-to-left language support
+            if (CultureInfo.CurrentUICulture.TextInfo.IsRightToLeft)
+            {
+                this.Body.Attributes.Add("class", "rtl ");
+            }
+

```

Comment:
I would prefer a solution where it is not as specific to a single culture.
Like `CultureInfo.CurrentUICulture.TextInfo.IsRightToLeft`

`this.Body.Attributes.Add` will replace the attribute, we need to get its current value, append the class and then .Add the attribute again.

Reply:
I think (but have not tried it) this line of the web.config defines the fallback language for when DNN can't know the language (which may be the case for 404 pages).

`<globalization culture="en-US" uiCulture="en" requestEncoding="UTF-8" responseEncoding="UTF-8" fileEncoding="UTF-8" />`

If that is not the case then I guess we would need to figure out for the 404 page. If we know the portal info in that context we could use the portal primary language (which could be a different PR maybe.

Your first change looks good to me if you can adjust the PR, I am fine with the described changes there.